src/include/migraphx/program.hpp

```diff
@@ -51,6 +51,10 @@ struct program_impl;
 
 struct marker;
 
+std::vector<argument> generic_eval(const migraphx::program& p,
```

Comment:
Make this a member function and just call it `eval`.

Reply:
Done.